x-pack/filebeat/input/gcppubsub/input.go

```diff
@@ -250,3 +245,24 @@ func (in *pubsubInput) getOrCreateSubscription(ctx context.Context, client *pubs
 
 	return nil, errors.New("no subscription exists and 'subscription.create' is not enabled")
 }
+
+func (in *pubsubInput) newPubsubClient(ctx context.Context) (*pubsub.Client, error) {
+	opts := []option.ClientOption{option.WithUserAgent(useragent.UserAgent("Filebeat"))}
+
+	if in.AlternativeHost != "" {
+		// this will be typically set because we want to point the input to a testing pubsub emulator
+		conn, err := grpc.Dial(in.AlternativeHost, grpc.WithInsecure())
+		if err != nil {
+			return nil, fmt.Errorf("grpc.Dial: %w", err)
```

Comment:
can we make this error more descriptive? "cannot connect to alternative_host" or similar